gammapy/modeling/models/core.py

```diff
@@ -428,6 +430,18 @@ def read(cls, filename):
     def from_yaml(cls, yaml_str, path=""):
         """Create from YAML string."""
         data = yaml.safe_load(yaml_str)
+        if "metadata" in data.keys():
+            creator_version = data["metadata"]["creator"].split()[1]
+            if check_version(creator_version) < 0:
```

Comment:
If there are other places where a check on the version is to be performed, this might be defined as a method on `CreatorMetaData` .